src/tech/v3/dataset/impl/column.clj

```diff
@@ -263,20 +313,26 @@
     (let [idx-rdr (->efficient-reader idx-rdr)]
       (if (== 0 (dtype/ecount missing))
         ;;common case
-        (Column. (->bitmap) (dtype/indexed-buffer idx-rdr data)
-                 metadata
-                 nil)
+        (let [bitmap (->bitmap)
+              new-data (dtype/indexed-buffer idx-rdr data)]
+          (Column. bitmap
+                   new-data
+                   metadata
+                   nil
+                   (delay (make-index-structure data bitmap))))
         ;;Uggh.  Construct a new missing set
         (let [idx-rdr (dtype/->reader idx-rdr)
               n-idx-elems (.lsize idx-rdr)
               ^RoaringBitmap result-set (->bitmap)]
           (dotimes [idx n-idx-elems]
-           (when (.contains missing (.readLong idx-rdr idx))
-             (.add result-set idx)))
-          (Column. result-set
-                   (dtype/indexed-buffer idx-rdr data)
-                   metadata
-                   nil)))))
+            (when (.contains missing (.readLong idx-rdr idx))
+              (.add result-set idx)))
+          (let [new-data (dtype/indexed-buffer idx-rdr data)]
+            (Column. result-set
+                     data
+                     metadata
+                     nil
+                     (delay (make-index-structure data result-set))))))))
   (to-double-array [col error-on-missing?]
```

Comment:
I think you meant to use `new-data` here in the column constructor -- not `data` for both the column constructor and for the index structure.

Reply:
Thank you. Good catch.